src/Compilers/Core/Portable/CommandLine/MergedAnalyzerConfig.cs

```diff
@@ -0,0 +1,196 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+using static Microsoft.CodeAnalysis.AnalyzerConfig;
+using AnalyzerOptions = System.Collections.Immutable.ImmutableDictionary<string, string>;
+using TreeOptions = System.Collections.Immutable.ImmutableDictionary<string, Microsoft.CodeAnalysis.ReportDiagnostic>;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Represents a set of <see cref="AnalyzerConfig"/> that can be applied to a given source location.
+    /// </summary>
+    public sealed class MergedAnalyzerConfig
```

Comment:
1. Why not a struct?
2. Let's call this `AnalyzerConfigSet`
3. The doc comment isn't that useful. Maybe mention something about how it represents all the AnalyzerConfigs in a given directory tree?

Reply:
We decided to leave this as a class, but we renamed this to AnalyzerConfigSet and I've reworded the doc comment to actually try to be helpful. (Novel, I know.)